File: 백준/Silver/2217. 로프/로프.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main(void) {
    int n;
    cin >> n;
    
    int *max_weight = new int[n];
    for (int i = 0; i < n; i++) {
        cin >> max_weight[i];
    }
    
    sort(max_weight, max_weight + n);
    int result = 0;
    
    result = max_weight[0] * n;
    for (int i = 1; i < n; i++) {
        if (result < max_weight[i] * (n - i)) {
            result = max_weight[i] * (n - i);
        }
    }
    cout << result << '\n';
    return 0;
}

File: 백준/Silver/2217. 로프/로프.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.Comparator;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        
        int min = Integer.MAX_VALUE;
        
        Integer[] weight = new Integer[n];
        for (int i = 0; i < n; i++) {
            weight[i] = scanner.nextInt();
        }
        
        Arrays.sort(weight, Comparator.reverseOrder());
        
        int max_weight = 0;
        for (int i = 0; i < weight.length; i++) {
        	int temp = weight[i] * (i + 1);
        	if (max_weight < temp) {
        		max_weight = temp;
        	}
        }
        scanner.close();
        System.out.println(max_weight);
    }
}